tests/bsim/bluetooth/host/gatt/device_name/src/common.h

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright (c) 2025 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_TESTS_BSIM_BLUETOOTH_HOST_GATT_DEVICE_NAME_SRC_COMMON_H_
+#define ZEPHYR_TESTS_BSIM_BLUETOOTH_HOST_GATT_DEVICE_NAME_SRC_COMMON_H_
+
+#include <zephyr/bluetooth/bluetooth.h>
+
+#define ADVERTISER_NAME      "Advertiser Pro II"
+#define SERVER_ORIGINAL_NAME "Server Super Name"
+
+static char server_new_name[CONFIG_BT_DEVICE_NAME_MAX];
+
+static void generate_name(char *name, size_t length)
+{
+	for (int i = 0; i < length; i++) {
+		name[i] = (i % 26) + 97;
+	}
+}
+
```

Comment:
Since you are defining `server_new_name` as a `char`, shouldn't you make room for and add the NULL terminator? That would make it possible to use `strlen(server_new_name)` instead, and treat it as a string here in the test

Reply:
No, my goal here is to check that even with a non-null terminated string the GATT write work fine. I should maybe have been more explicit 😅

Reply:
> No, my goal here is to check that even with a non-null terminated string the GATT write work fine. I should maybe have been more explicit 😅

That should still work as expected when `server_new_name` is NULL terminated. When do you perform the write and use `strlen` then you'll only send the string as non-NULL terminated over air. 

My point was to avoid having a non-NULL terminated string value stored in a `char` array.

Reply:
I see what you mean, would it be better to have `server_new_name` be a byte array then?

Reply:
> I see what you mean, would it be better to have server_new_name be a byte array then?

That is one option, yeah :) 

I don't see any issues storing it as a NULL-terminated string though. It doesn't affect the test, but allows the test to use string functions.

Reply:
I updated to use `uint8_t`. It's just a personal opinion but I feel the message of the test is better communicated without the NULL char